python/lsst/ts/scheduler/telemetry_stream_handler.py

```diff
@@ -0,0 +1,270 @@
+# This file is part of ts_scheduler
+#
+# Developed for the LSST Telescope and Site Systems.
+# This product includes software developed by the LSST Project
+# (https://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+
+__all__ = ["TelemetryStreamHandler"]
+
+import logging
+import numpy
+
+from typing import Any, List, Dict
+
+from astropy import units
+from astropy.time import Time, TimeDelta
+
+
+from lsst.ts import salobj
+
+from .utils import get_efd_client
+from . import CONFIG_SCHEMA
+
+
+class TelemetryStreamHandler:
+    """Handle telemetry stream.
+
+    Parameters
+    ----------
+    log : `logging.Logger`
+        Logger class.
+    efd_name : `str`
+        Name of the efd instance to query telemetry from.
+    """
+
+    def __init__(
+        self,
+        log: logging.Logger,
+        efd_name: str,
+    ) -> None:
+
+        self.log = log.getChild(type(self).__name__)
+
+        self.efd_name: str = efd_name
+        self.telemetry_streams: Dict = dict()
+
+        self.efd_client: Any = None
+
+    async def configure_telemetry_stream(self, telemetry_stream) -> None:
+        """Configure telemetry streams.
+
+        Parameters
+        ----------
+        telemetry_stream : `list` of `dict`
+            List of telemetry streams to parse.
+
+        Raises
+        ------
+        ValidationError:
+            If one (or more) stream is invalid.
+        RuntimeError:
+            If selected efd_table are not in the selected EFD instance.
+            If selected efd_columns are not in the topic attributes.
+        """
+
+        self._configure_efd_client()
+
+        validated_telemetry_stream = self._get_validated_telemetry_stream(
+            telemetry_stream
+        )
+
+        await self._check_efd_tables(validated_telemetry_stream)
+
+        await self._check_efd_columns(validated_telemetry_stream)
+
+        self.telemetry_streams = validated_telemetry_stream
+
+    def _configure_efd_client(self):
+        """Configure EFD client."""
+        if self.efd_client is None:
+            self.log.info(f"Initializing efd client for {self.efd_name} instance.")
+            self.efd_client = get_efd_client(self.efd_name)
+
+    def _get_validated_telemetry_stream(self, telemetry_stream: List[dict]) -> Dict:
```

Comment:
`List[dict]` -> `List[Dict]`. MyPy should complain about using `dict` here.

EDIT: I see you did not update setup.cfg to make sure that pytest calls mypy as well, which is why this wasn't caught. Please consider doing that in a future ticket.